blockindex/indexbuilder.go

```diff
@@ -129,19 +129,19 @@ func (ib *IndexBuilder) init() error {
 		return err
 	}
 	// update index to latest block
+	blks := make([]*block.Block, 0, 5000)
 	for startHeight++; startHeight <= tipHeight; startHeight++ {
 		blk, err := ib.dao.GetBlockByHeight(startHeight)
 		if err != nil {
 			return err
 		}
-		if err := ib.indexer.PutBlock(blk); err != nil {
-			return err
-		}
+		blks = append(blks, blk)
 		// commit once every 5000 blocks
 		if startHeight%5000 == 0 || startHeight == tipHeight {
-			if err := ib.indexer.Commit(); err != nil {
+			if err := ib.indexer.PutBlocks(blks); err != nil {
```

Comment:
don't think this is a good design API

this could cause significant memory throttle and may slow down the full-node sync